File: 3043. Find the Length of the Longest Common Prefix/solution.cpp
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {

        unordered_set<string>st;
        int n1 = arr1.size(), n2 = arr2.size(),mx=0;

        for(int i=0;i<n1;i++){
            string curr = "";
            string str = to_string(arr1[i]);
            for(auto &x : str){
                curr+=x;
                st.insert(curr);
            }      
        }

        for(int i=0;i<n2;i++){
            string str = to_string(arr2[i]);
            string curr = "";

            for(auto &x : str){
                curr+= x;
                if(st.count(curr)){
                    mx = max(mx,(int)curr.length());
                }
            }
        }
        
        return mx;
        
    }
};

File: 3043. Find the Length of the Longest Common Prefix/solution.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number}
 */
var longestCommonPrefix = function(arr1, arr2) {

    let st = new Set();
    let n1 = arr1.length, n2 = arr2.length, mx = 0;

    for (let i = 0; i < n1; i++) {
        let curr = "";
        let str = arr1[i].toString();
        for (let x of str) {
            curr += x;
            st.add(curr);
        }
    }

    for (let i = 0; i < n2; i++) {
        let curr = "";
        let str = arr2[i].toString();
        for (let x of str) {
            curr += x;
            if (st.has(curr)) {
                mx = Math.max(mx, curr.length);
            }
        }
    }

    return mx;   
};